core/src/avm2/optimize.rs

```diff
@@ -401,8 +401,7 @@ pub fn optimize<'gc>(
             }
             Op::CoerceI => {
                 let stack_value = stack.pop_or_any();
-                // TODO: maybe the type check is safe here...?
-                if stack_value.contains_valid_integer {
+                if stack_value.class == Some(types.int) {
```

Comment:
This prevents optimizing `pushuint 5; coerce_i`, right? Could we check "type is int or value is statically known to be integral" instead?